pkg/model/mask.go

```diff
@@ -115,6 +110,33 @@ func updateField(src, dst reflect.Value, fields []string) {
 	}
 }
 
+// updateSliceField this function updates a destination slice by either replacing a struct with a matching ID
+// or appending new values if no matching ID is found.
+func updateSliceField(srcField, dstField reflect.Value) {
```

Comment:
Maybe we should consider using a custom implementation for update_mask 'messages' since it might be used frequently.

Reply:
When we have a slice with a struct element with `ID` field, we assume it is the unique identifier and we can update this element.

What update mask do we use to update the Message in Translation? `messages.*`?

Reply:
https://github.com/protocolbuffers/protobuf/blob/9bbea4aa65bdaf5fc6c2583e045c07ff37ffb0e7/src/google/protobuf/field_mask.proto#L111

```proto
// If a repeated field is specified for an update operation, new values will
// be appended to the existing repeated field in the target resource. Note that
// a repeated field is only allowed in the last position of a `paths` string.
```

I assume, that when the update mask is a repeated field, we append by default. However, if the repeated message contains the field `id` (unique identifier), we replace it, right?

In our case, update mask `messages` replaces the message by its id:
* appends if it is a new message
* replaces if it is an existing message

Reply:
@janishorsts Currently the updatesSliceField function appends a new message if ID doesn't exist and replaces a message if ID exists. You can check tests in mask_test.go. So as I understood it is expected behavior, right?